File: app/src/main/java/com/example/rwothoromo/developers/model/GithubUsersResponse.java
package com.example.rwothoromo.developers.model;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * This class gets the GitHub API text response and parses it to JSON.
 */
public class GithubUsersResponse {

    @SerializedName("items")
    private List<GithubUser> githubUsers;

    /**
     * Returns GithubUsersResponse in JSON format.
     *
     * @param response the API response
     * @return GithubUsersResponse the GSON reponse
     */
    public static GithubUsersResponse parseJSON(String response) {
        Gson gson = new GsonBuilder().create();
        return gson.fromJson(response, GithubUsersResponse.class);
    }

    /**
     * Returns the GitHub user list.
     *
     * @return githubUsers List
     */
    public List<GithubUser> getResult() {
        return githubUsers;
    }
}

File: app/src/main/java/com/rwothoromo/developers/model/GithubUsersResponse.kt
package com.rwothoromo.developers.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

/**
 * This class gets the GitHub API text response and parses it to JSON.
 */
class GithubUsersResponse {

    /**
     * Returns the GitHub user list.
     *
     * @return githubUsers List
     */
    @SerializedName("items")
    val result: List<GithubUser>? = null

    companion object {

        /**
         * Returns GithubUsersResponse in JSON format.
         *
         * @param response the API response
         * @return GithubUsersResponse the GSON reponse
         */
        fun parseJSON(response: String): GithubUsersResponse {
            val gson = GsonBuilder().create()
            return gson.fromJson(response, GithubUsersResponse::class.java)
        }
    }
}
